### PR TITLE
test: add optional nock calls

### PR DESCRIPTION
**Pull request by Realtin**
because if we use the same installation ids then the githubqueue
allready exists and won't need to make some of these requests!

### CODE REVIEW

**Comment by gr2m** [comment]
ohh good catch! 

Does that mean we reuse installation ids across tests? Maybe you can avoid that to test in isolation? Or did I misunderstand the problem?

**Comment by Realtin** [comment]
@gr2m yep, not reusing the installation ids would be another solution. 
When writing some of the newer tests we probably just reused installations because we didn't anticipate that this would be a problem.

**Comment by gr2m** [comment]
What I often times do in such cases is creating a file like `./test/unique-job-number.js` which looks something like this

```js
module.exports = uniqueJobNumber

let currentNumber = 0

uniqueJobNumber () {
  return ++currentNumber
}
```

and then use the `uniqueJobNumber()` function in all tests so the code is always the same and you always get a unique number each time you call the function

File: test/jobs/create-version-branch.js [hunk 1]
```diff
@@ -4,6 +4,9 @@ const proxyquire = require('proxyquire').noCallThru()
 
 const dbs = require('../../lib/dbs')
 
+nock.disableNetConnect()
+nock.enableNetConnect('localhost')
```
Comment (rmehner) [inline test/jobs/create-version-branch.js]: I wonder why we don't have this for all our tests, we actually never want to call GitHub by accident. Good catch 👍

**Pull request merged**